File: diario-classe-backend/src/main/java/com/projetointegrador/diarioclasse/entity/Acesso.java
package com.projetointegrador.diarioclasse.entity;

import com.projetointegrador.diarioclasse.enums.Role;
import com.projetointegrador.diarioclasse.enums.TipoAcesso;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tb_acessos")
public class Acesso {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // ID da pessoa (aluno, professor, etc.)
    @Column(nullable = false)
    private Long pessoaId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Role role;

    private LocalDateTime data;

    @Enumerated(EnumType.STRING)
    private TipoAcesso tipo;
}

File: diario-classe-frontend/src/models/Acesso.ts
import {Roles} from "../enums/Roles.ts";
import {TiposAcessos} from "../enums/TiposAcessos.ts"

export interface Acesso {
  id: number;
  pessoaId: number;
  nome: string;
  role?: Roles[];
  dataHora: string;
  tipo?: TiposAcessos[];
}